bus.py

```diff
@@ -0,0 +1,97 @@
+import random
+from utility import distance, firstn, position_on_edge
+
+
+def _road_point_generator(roads):
+    assert len(roads) > 0
+
+    # Length of all roads combined
+    total_length = sum(map(lambda road: road.getLength(), roads))
+
+    while True:
+        # Select a point on the combined stretch of road
+        distance = random.uniform(0, total_length)
+
+        # Find the selected road
+        length_sum = 0.0
+        found_road = False
+        for road in roads:
+            from_node = road.getFromNode().getCoord()
+            to_node = road.getToNode().getCoord()
+
+            if (length_sum + road.getLength()) >= distance:
+                # Compute the exact point on the selected road
+
+                # Distance along the road segment
+                remaining = length_sum - distance
+                pos = position_on_edge(road, remaining)
+
+                yield [
+                        pos[0],
+                        pos[1],
+                        road,
+                        remaining]
+                found_road = True
+                break
+            else:
+                length_sum += road.getLength()
+
+        if not found_road:
+            raise AssertionError("Failed to pick a road. A distance beyound the last road must have been erroneously picked: {} (length sum: {}) (total length: {})".format(distance, length_sum, total_length))
+
+
+def bus_stop_generator(roads, inner_r, outer_r, k=10, seeds=[]):
+    """
+    Bus stop placement using the poisson-disc algorithm
+    """
+    assert inner_r < outer_r
+
+    all_points = list(seeds)
+
+    road_points_gen = _road_point_generator(roads)
+    if len(all_points) <= 0:
+        road = tuple(next(road_points_gen))
+        print(road)
+        yield road
+        all_points.add(road) # Seed point
+
+    active_points = list(all_points) # Use a list because random.choice require a sequence
+
+    def check_dist(p, test_points, limit=inner_r):
```

Comment:
I would personally invert the logic of this or rename it. Currently it sounds like it "checks if point is okay" instead of "checks if point is NOT okay".

Reply:
I disagree, previously I had it reversed but that became to confusing

Reply:
You sure? You are inverting everywhere already.
Example L80-86 in bus.py:
```python
# ...
try:
    # Search for a point, or raise StopIteration is none can be found
    point = next(filter(
            lambda p: not check_dist(p, iter(all_points)),
            gen))

    assert not check_dist(point, all_points)
# ...
```

Reply:
I wrote it the way I found most logical, and easy to read. Is there any problem with function of the code, otherwise we are just pointlessly arguing over wording.